### PR TITLE
Run `awesome-lint` on new additions

### PR DESCRIPTION
**Pull request by RayBB**
Note: This PR is a repeat of #1851 since the fork or original fork was deleted. This PR address comments there.

I really appreciate this project and wanted to try to give back so it's a little easier for everyone involved.
To that end, I created this PR to **automatically lint new lists** when someone opens a PR to add them.

## Benefits
  - It will save time of maintainers by allowing them to look at the error message first
  - It will provide immediate feedback to people who make a PR
  - It helps people having trouble installing `awesome-lint`
  - It is free

## Risks
  - As far as I know, there is no risk to users or maintainers.

## What it does
- PR will lint **NEW** lists that are requesting to be added.
- PR will **NOT** lint [this](https://github.com/sindresorhus/awesome/) repo itself. If that functionality is desired it can happen in a separate PR (see https://github.com/sindresorhus/awesome/pull/1794).
- PR will **NOT** lint the current lists in the readme.

## How it does it
This PR creates a GitHub action that will do the following every time a new commit is pushed to an open PR:
1. Find the URL that is being added using `git diff`
2. Clone that repo
3. Run `awesome-lint`on that repo.

The above 3 steps will **ONLY** run if `readme.md` is changed.
If readme is changed and no new URL is detected in the readme it will NOT fail with a helpful error message.
Error can be ignored if it is a cosmetic PR

[Here](https://github.com/RayBB/awesome/pull/2) are examples of failing commits.

## Final notes

"Squash and Merge" this PR or let me know and I will squash it manually.

Feel free to edit this any way you see fit or ask me to make the edits. 👍
If there are any questions on how this works drop them here and I'll try my best to answer.

Again, thanks so much for all the hard work the maintainers put into this repo! 👏

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# This is a basic workflow for github actions
```
Comment (sindresorhus) [inline .github/workflows/main.yml:1]: Remove all the comments and please follow the format of https://github.com/sindresorhus/awesome/pull/1794/files

File: .github/workflows/repo_linter.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+# by default github actions doesn't fetch the whole repo so we are getting the latest commit on main
+git fetch origin main:main --depth=1
```
Comment (sindresorhus) [inline .github/workflows/repo_linter.sh:4]: This can be set in the workflow file.

Comment (RayBB) [reply .github/workflows/repo_linter.sh:4]: I didn't know that. Thanks for sharing that's very handy.

File: .github/workflows/main.yml (hunk 1 above)
Comment (RayBB) [reply .github/workflows/main.yml]: @sindresorhus, I removed the comments and added fetch depth. I'm not if you wanted anything else under the "follow the format" category but I: 1. changed the job from `run_awesome_lint` to `lint` and removed the top level "name". 

If there is anything else you want changed feel free to change yourself or tag me when you respond and I'll fix asap.

**Comment by RayBB** [comment]
@sindresorhus have I addressed your concerns?

**Comment by sindresorhus** [comment]
Thanks :)

**Comment by sindresorhus** [comment]
@RayBB The scripts throws an error: https://github.com/sindresorhus/awesome/pull/1990/checks?check_run_id=2468601026

**Review by Tolito20 — COMMENTED** [review]
Galván Gómez

**Comment by HOP01** [comment]
Ok

**Pull request merged**